### PR TITLE
Support span labels with the plugin details for the SimpleBuildStep implementations

### PR DESCRIPTION
**Pull request by v1v**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### What

Some plugins implement the [SimpleBuildStep](https://javadoc.jenkins.io/jenkins/tasks/SimpleBuildStep.html) and therefore there is no way to access directly to the pluginwrapper context for the step itself.

Those steps use the description class https://javadoc.jenkins.io/plugin/workflow-basic-steps/org/jenkinsci/plugins/workflow/steps/CoreStep.html and their `symbol`, aka their step syntax, can be accessed through the `ArgumentsAction` class, and more precisely the `delegate` key.  See [mailing-list](https://groups.google.com/g/jenkinsci-dev/c/UbJAeh8UZnE/m/KE5UIECOAAAJ)

### Test

For instance see the execution for the `googleStorageUpload` step provided by https://github.com/jenkinsci/google-storage-plugin/blob/6618c8ef7c4a0eaf2f2bc8a0b48383667e2478e3/src/main/java/com/google/jenkins/plugins/storage/ClassicUploadStep.java

image

### Issue

Fixes https://github.com/jenkinsci/opentelemetry-plugin/issues/133

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java [hunk 1]
```diff
@@ -228,18 +235,31 @@ public void addStepPlugin(String stepName, StepPlugin c) {
...
 
+    @Nullable
+    private Descriptor<? extends Describable> getStepDescriptor(@Nonnull FlowNode node, @Nullable Descriptor<? extends Describable> descriptor) {
+        // Support for https://javadoc.jenkins.io/jenkins/tasks/SimpleBuildStep.html
+        if (descriptor instanceof CoreStep.DescriptorImpl) {
+            Map<String, Object> arguments = ArgumentsAction.getFilteredArguments(node);
+            UninstantiatedDescribable describable = (UninstantiatedDescribable) arguments.get("delegate");
+            if (describable != null) {
+                return SymbolLookup.get().findDescriptor(Describable.class, describable.getSymbol());
+            }
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java:246]:
```suggestion
            if(arguments.get("delegate") instanceOf UninstantiatedDescribable){
              UninstantiatedDescribable describable = (UninstantiatedDescribable) arguments.get("delegate");
              if (describable != null) {
                  return SymbolLookup.get().findDescriptor(Describable.class, describable.getSymbol());
              }
            }
```

File: src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java [hunk 2]
```diff
@@ -209,26 +215,45 @@ private boolean isIgnoredStep(@Nullable StepDescriptor stepDescriptor) {
...
+        }
         return stepDescriptor.getFunctionName();
     }
 
+    @Nullable
+    private UninstantiatedDescribable getUninstantiatedDescribableOrNull(@Nonnull FlowNode node, @Nullable StepDescriptor stepDescriptor) {
+        // Support for https://javadoc.jenkins.io/jenkins/tasks/SimpleBuildStep.html
+        if (stepDescriptor instanceof CoreStep.DescriptorImpl) {
+            Map<String, Object> arguments = ArgumentsAction.getFilteredArguments(node);
+            return (UninstantiatedDescribable) arguments.get("delegate");
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java:246]:
```suggestion
            if(arguments.get("delegate") instanceOf UninstantiatedDescribable){
              return (UninstantiatedDescribable) arguments.get("delegate");
            }
```

**Review by kuisathaverat — APPROVED** [review]

File: src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java [hunk 3]
```diff
@@ -209,26 +215,45 @@ private boolean isIgnoredStep(@Nullable StepDescriptor stepDescriptor) {
         return ignoreStep;
     }
 
-    private String getStepName(@Nullable StepDescriptor stepDescriptor, @Nonnull String name) {
+    private String getStepName(@Nonnull FlowNode node, @Nullable StepDescriptor stepDescriptor, @Nonnull String name) {
```
Comment (cyrille-leclerc) [inline src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java:218]: why not removing the `stepDescriptor` parameter is is obtained doing a `node.getDescriptor()`?

Comment (v1v) [reply src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java:218]: Unfortunately, some of the `onStart` have a different object that did not implement the `getDescriptor()` so I decided to pass the argument instead, but let me confirm it

Comment (v1v) [reply src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java:218]: For the `getStepName` is possible but for the `getTypeName`, I'm afraid there are two type of consumers:

- `io.jenkins.plugins.opentelemetry.job.MonitoringPipelineListener#onStartNodeStep` with the  [`StepStartNode`](https://javadoc.jenkins.io/plugin/workflow-cps/org/jenkinsci/plugins/workflow/cps/nodes/StepStartNode.html) argument but it does implement `org.jenkinsci.plugins.workflow.cps.nodes.StepStartNode#getDescriptor`
- `io.jenkins.plugins.opentelemetry.job.MonitoringPipelineListener#onAtomicStep` with the [`StepAtomNode`](https://javadoc.jenkins.io/plugin/workflow-cps/org/jenkinsci/plugins/workflow/cps/nodes/StepAtomNode.html) argument that overrides `org.jenkinsci.plugins.workflow.graph.StepNode#getDescriptor`

I could not find any interface or super-class in common for `StepStartNode` and `StepAtomNode` , so I've decided to add a new parameter to the signature.

Comment (cyrille-leclerc) [reply src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java]: 👍

**Review by cyrille-leclerc — APPROVED** [review]

**Pull request merged**